src/node.cc

```diff
@@ -132,47 +148,100 @@ TSNode UnmarshalNode(const Tree *tree) {
   return result;
 }
 
-static void ToString(const Nan::FunctionCallbackInfo<Value> &info) {
+namespace {
+void ToString(const Nan::FunctionCallbackInfo<Value> &info) {
   const Tree *tree = Tree::UnwrapTree(info[0]);
   TSNode node = UnmarshalNode(tree);
-  if (node.id) {
+
+  if (node.id != nullptr) {
     const char *string = ts_node_string(node);
     info.GetReturnValue().Set(Nan::New(string).ToLocalChecked());
-    free((char *)string);
+    free(const_cast<char *>(string));
   }
 }
 
-static void IsMissing(const Nan::FunctionCallbackInfo<Value> &info) {
+void IsMissing(const Nan::FunctionCallbackInfo<Value> &info) {
   const Tree *tree = Tree::UnwrapTree(info[0]);
   TSNode node = UnmarshalNode(tree);
-  if (node.id) {
+
+  if (node.id != nullptr) {
     bool result = ts_node_is_missing(node);
     info.GetReturnValue().Set(Nan::New<Boolean>(result));
   }
 }
 
-static void HasChanges(const Nan::FunctionCallbackInfo<Value> &info) {
+void IsExtra(const Nan::FunctionCallbackInfo<Value> &info) {
+  const Tree *tree = Tree::UnwrapTree(info[0]);
+  TSNode node = UnmarshalNode(tree);
+
+  if (node.id != nullptr) {
+    bool result = ts_node_is_extra(node);
+    info.GetReturnValue().Set(Nan::New<Boolean>(result));
+  }
+}
+
+void HasChanges(const Nan::FunctionCallbackInfo<Value> &info) {
   const Tree *tree = Tree::UnwrapTree(info[0]);
   TSNode node = UnmarshalNode(tree);
-  if (node.id) {
+
+  if (node.id != nullptr) {
     bool result = ts_node_has_changes(node);
     info.GetReturnValue().Set(Nan::New<Boolean>(result));
   }
 }
 
-static void HasError(const Nan::FunctionCallbackInfo<Value> &info) {
+void HasError(const Nan::FunctionCallbackInfo<Value> &info) {
   const Tree *tree = Tree::UnwrapTree(info[0]);
   TSNode node = UnmarshalNode(tree);
-  if (node.id) {
+
+  if (node.id != nullptr) {
     bool result = ts_node_has_error(node);
     info.GetReturnValue().Set(Nan::New<Boolean>(result));
   }
 }
 
-static void FirstNamedChildForIndex(const Nan::FunctionCallbackInfo<Value> &info) {
+void IsError(const Nan::FunctionCallbackInfo<Value> &info) {
+  const Tree *tree = Tree::UnwrapTree(info[0]);
+  TSNode node = UnmarshalNode(tree);
+
+  if (node.id != nullptr) {
+    bool result = ts_node_is_error(node);
+    info.GetReturnValue().Set(Nan::New<Boolean>(result));
+  }
+}
+
+void ParseState(const Nan::FunctionCallbackInfo<Value> &info) {
+  const Tree *tree = Tree::UnwrapTree(info[0]);
+  TSNode node = UnmarshalNode(tree);
+
+  if (node.id != nullptr) {
+    info.GetReturnValue().Set(Nan::New(ts_node_parse_state(node)));
```

Comment:
shouldn't these have a type?

Reply:
parse_state gets the parse state (a number) of a node, Nan::New defaults to int32_t iirc